README.md

```diff
@@ -204,6 +204,100 @@ end
 Using `insert/2` in factory definitions may lead to performance issues and bugs,
 as records will be saved unnecessarily.
 
+## Lazy Attributes
+
+You may defer calculation of your attributes, and even base one attribute upon
+another, by using functions:
+
+```elixir
+def hero_factory do
+  %{
+    creature: "Bat",
+    nickname: &"#{&1.creature}man",
+    vehicle: &"#{&1.creature}mobile"
+  }
+end
+
+build(:hero)
+# => %{creature: "Bat", nickname: "Batman", vehicle: "Batmobile"}
+
+build(:hero, creature: "Spider", vehicle: "webs")
+# => %{creature: "Spider", nickname: "Spiderman", vehicle: "webs"}
+```
+
+This is especially helpful in complex object graphs where a record and its
+child both belong to the same parent.  This is common in multi-tenant apps,
+where many of the records belong to the same scope.  Consider a multi-site blog
+platform where Articles and Authors both belong to a Site, and Articles also
+belong to the User:
+
+```txt
+Site <---- Author
+  ^          ^
+  |          |
+  +------- Article
+```
+
+Such factories would be:
+
+```elixir
+def site_factory do
+  %Site{
+    name: "Awesome Blog"
+  }
+end
+
+def author_factory do
+  %Author{
+    name: "Danny Tanner",
+    email: "danny@example.com",
+    site: build(:site)
+  }
+end
+
+def article_factory do
+  %Article{
+    title: "The House is Full",
+    site: create(:site), # Unfortunately, site creation can't be deferred or Author won't see it
+    author: build(:author, site: &(&1.site))
```

Comment:
The above is sub-optimal, but I don't know how to solve it yet.  I'd prefer to be able to do this:

```ex
site: &create(:site), # Defer creation until we know we need it, so it isn't created if it will be overwritten by attributes
author: build(:author, site: &(&1.site))
```

But there's a problem: if `author` gets resolved first, then `&1.site` will return a function instead of a site - there's a circular dependency.  Maybe we should just add a warning in the README to avoid circular dependencies?